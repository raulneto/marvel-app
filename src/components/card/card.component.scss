.comics-card {
    cursor: pointer;
    position: relative;
    height: 252px;
    margin: 10px;
    width: 168px;
    overflow: hidden;

    &:after {
        content: '';
        border: 8px solid transparent;
        width: 100%;
        position: absolute;
        left: 0;
        top: 0;
        height: 100%;
        box-sizing: border-box;
        transition: border-color 0.5s;
    }
    
    &.is-fav {
        &:after {
            border-color: #ff0000;
        }
    }

    &:hover {
        .comic-img {
            opacity: .5;
            transition: opacity 0.5s;
        }
        .info {
            transform: translateY(0px);
        }
        .favourite {
            .icon {
                background-image: url('../../assets/images/heart_off.png');
            }
        }
    }
    

    .info {
        box-sizing: border-box;
        border-bottom: 8px solid #ff0000;
        position: absolute;
        background-color: #fff;
        bottom: 0;
        padding: 4px 15px;
        width: 100%;
        transform: translateY(100px);
        transition: transform 0.5s;

        .text {
            font-weight: bold;
            font-size: 12px;
            line-height: 16px;
            text-transform: uppercase;
        }
    }

    .favourite {
        height: 37px;
        width: 40px;
        position: absolute;
        right: 10px;
        display: flex;
        top: 10px;
        align-items: center;
        justify-content: center;
        background: transparent;
        border: 0 none;
        padding: 0;
        cursor: pointer;

        .icon {
            background-repeat: no-repeat;
            height: 27px;
            width: 30px;
            z-index: 1;
        }

        &:hover {
            .icon {
                background-image: url('../../assets/images/heart_hover.png');
            }
        }
    }

    &.is-fav {
        .favourite {
            .icon {
                background-image: url('../../assets/images/heart_on.png');
                height: 37px;
                width: 40px;
            }
        }
    }

    &.placeholder {
        animation: placeholder-pulse 1.5s infinite;
        background-color: rgb(205, 205, 205);
    }
}

@keyframes placeholder-pulse {
    0%   { opacity: .6; }
    50% { opacity: 1; }
    100% { opacity: .6; }
}