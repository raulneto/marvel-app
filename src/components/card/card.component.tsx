import * as React from "react";
import './card.component.scss';

type ICard = {
    id: string;
    image: any;
    isFavourite: boolean;
    setFavouriteComic: (id: string, isFavourite: boolean) => void;
    title: string;
}

export const CardComponent: React.FC<ICard> = (props) =>  {
    const { id, image, isFavourite, setFavouriteComic, title } = props;
    const favouriteClass = isFavourite ? 'is-fav' : '';

    return (
        <div className={`comics-card ${favouriteClass}`}>
            <img alt={title} className="comic-img" src={image} />
            <button onClick={() => setFavouriteComic(id, isFavourite)} className="favourite">
                <span className="icon"></span>
            </button>
            <div className="info">
                <div className="text">{title}</div>
            </div>
        </div>
    );
}

export const CardPlaceHolder: React.FC = () => {
    return <div className="comics-card placeholder"></div>
}