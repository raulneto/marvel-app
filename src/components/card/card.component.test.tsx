import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { CardComponent } from './card.component';
import { QueryClient, QueryClientProvider } from 'react-query';

const queryClient = new QueryClient();

test('CardComponent render', () => {
    const mockFavouriteComicFn = jest.fn();
	const component = shallow(
        <CardComponent 
            id="123" 
            image="image" 
            isFavourite={false}
            setFavouriteComic={mockFavouriteComicFn}
            title="title"
        />
    );

	expect(toJson(component)).toMatchSnapshot();
    
    component.find('.favourite').simulate('click');
    expect(mockFavouriteComicFn.mock.calls.length).toEqual(1);
});
