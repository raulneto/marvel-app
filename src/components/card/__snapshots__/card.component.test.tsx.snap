// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent render 1`] = `
<div
  className="comics-card "
>
  <img
    alt="title"
    className="comic-img"
    src="image"
  />
  <button
    className="favourite"
    onClick={[Function]}
  >
    <span
      className="icon"
    />
  </button>
  <div
    className="info"
  >
    <div
      className="text"
    >
      title
    </div>
  </div>
</div>
`;
